schematic/store/synapse.py

```diff
@@ -1917,14 +1974,16 @@ def upload_manifest_combo(
         )
 
         if file_annotations_upload:
-            manifest = self.add_annotations_to_entities_files(
-                dmge,
-                manifest,
-                manifest_record_type,
-                datasetId,
-                hideBlanks,
-                manifest_synapse_table_id,
-                annotation_keys=annotation_keys,
+            manifest = asyncio.run(
```

Comment:
Using asyncio.run will work (for most cases). The cases where this will not work is if you are running this from within a jupyter notebook. If this is a concern read below:

https://blog.jupyter.org/ipython-7-0-async-repl-a35ce050f7f7

The problem is that the notebook will already have an Asyncio event loop running, and you cannot using `asyncio.run` by default from within an event loop. To get around this in the python client I wrote this wrapper:

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/v4.3.0/synapseclient/core/async_utils.py#L80-L99

Here is an example of where I used that code:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/v4.3.0/synapseutils/sync.py#L1225-L1233

Reply:
I think we don't need to support async runs in jupyter notebook. But what you wrote is interesting! I saw that you used: `nest_asyncio.apply(loop=loop)`. Is the idea here that you are trying to add another async loop? And you will run both loops until they both complete by doing: `return loop.run_until_complete(coroutine)`?

Reply:
> nest_asyncio.apply(loop=loop). Is the idea here that you are trying to add another async loop?

Not exactly. This is attaching the new async function to the existing asyncio loop that is already running. Even if the code that you called this from isn't an `async def` function. It's a bit hacky, but the use-case is that we need to be able to support running in a notebook seamlessly.

> And you will run both loops until they both complete by doing: return loop.run_until_complete(coroutine)?

This will run until the coroutine I specifically passed to it is completed. This doesn't end the entrant asyncio loop.

Reply:
I see. Thanks for the explanation!